meshio/medit_io.py

```diff
@@ -68,47 +80,50 @@ def read_buffer(file):
 
         assert keyword.isalpha()
 
-        meshio_from_medit = {
-            "Edges": ("line", 2),
-            "Triangles": ("triangle", 3),
-            "Quadrilaterals": ("quad", 4),
-            "Tetrahedra": ("tetra", 4),
-            "Hexahedra": ("hexahedra", 8),
-        }
-
         if keyword == "MeshVersionFormatted":
-            assert reader.next_item() == "1"
+            version = reader.next_item()
+            dtype = {"1": c_float, "2": c_double}[version]
         elif keyword == "Dimension":
             dim = int(reader.next_item())
         elif keyword == "Vertices":
             assert dim > 0
             # The first value is the number of nodes
             num_verts = int(reader.next_item())
-            points = numpy.empty((num_verts, dim), dtype=float)
+            points = numpy.empty((num_verts, dim), dtype=dtype)
+            point_data["medit:ref"] = numpy.empty(num_verts, dtype=int)
             for k in range(num_verts):
-                # Throw away the label immediately
-                points[k] = numpy.array(reader.next_items(dim + 1), dtype=float)[:-1]
+                points[k] = numpy.array(reader.next_items(dim), dtype=dtype)
+                point_data["medit:ref"][k] = reader.next_item()
         elif keyword in meshio_from_medit:
             meshio_name, num = meshio_from_medit[keyword]
             # The first value is the number of elements
             num_cells = int(reader.next_item())
-            cell_data = numpy.empty((num_cells, num), dtype=int)
+            cell_data[meshio_name] = {
+                "gmsh:physical": numpy.empty(num_cells, dtype=int)
+            }
+            cells1 = numpy.empty((num_cells, num), dtype=int)
             for k in range(num_cells):
                 data = numpy.array(reader.next_items(num + 1), dtype=int)
-                # Throw away the label
-                cell_data[k] = data[:-1]
+                cells1[k] = data[:-1]
+                cell_data[meshio_name]["gmsh:physical"][k] = data[-1]
 
             # adapt 0-base
-            cells[meshio_name] = cell_data - 1
+            cells[meshio_name] = cells1 - 1
         else:
             assert keyword == "End", "Unknown keyword '{}'.".format(keyword)
 
-    return points, cells
+    return Mesh(points, cells, point_data=point_data, cell_data=cell_data)
 
 
 def write(filename, mesh):
     with open(filename, "wb") as fh:
-        fh.write(b"MeshVersionFormatted 1\n")
+        if mesh.points.dtype == c_float:
+            fh.write(b"MeshVersionFormatted 1\n")
+        elif mesh.points.dtype == c_double:
+            fh.write(b"MeshVersionFormatted 2\n")
+        else:
+            raise NotImplementedError("Unknown points.dtype '{}'.".format(
```

Comment:
Will never be tested. I prefer
```
if  mesh.points.dtype == c_float:
    # ...
else:
   assert mesh.points.dtype == c_double, "Unknown points.dtype '{}'.format(...)
```

Reply:
I liked the dict-lookup but no:
```
                     {c_float: 1,
>                     c_double: 2}[mesh.points.dtype])
E           KeyError: dtype('float64')
```
so I went with the `if`-`else`; I guess getitem uses a different kind of equality than `==`.